tuftool/src/update_targets.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::build_targets;
+use crate::datetime::parse_datetime;
+use crate::error::{self, Result};
+use crate::source::parse_key_source;
+use chrono::{DateTime, Utc};
+use snafu::ResultExt;
+use std::fs::File;
+use std::num::NonZeroU64;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tempfile::tempdir;
+use tough::editor::RepositoryEditor;
+use tough::http::HttpTransport;
+use tough::key_source::KeySource;
+use tough::{ExpirationEnforcement, FilesystemTransport, Limits, Repository};
+use url::Url;
+
+#[derive(Debug, StructOpt)]
+pub(crate) struct UpdateTargetsArgs {
+    /// Delegatee role
+    #[structopt(long = "role", required = true)]
+    role: String,
+
+    /// Key files to sign with
+    #[structopt(short = "k", long = "key", required = true, parse(try_from_str = parse_key_source))]
+    keys: Vec<Box<dyn KeySource>>,
+
+    /// Expiration of new role file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(short = "e", long = "expires", parse(try_from_str = parse_datetime))]
+    expires: Option<DateTime<Utc>>,
+
+    /// Version of targets.json file
+    #[structopt(short = "v", long = "version")]
+    version: Option<NonZeroU64>,
+
+    /// Path to root.json file for the repository
+    #[structopt(short = "r", long = "root")]
+    root: PathBuf,
+
+    /// TUF repository metadata base URL
+    #[structopt(short = "m", long = "metadata-url")]
+    metadata_base_url: Url,
+
+    /// Directory of targets
+    #[structopt(short = "t", long = "add-targets")]
+    targets_indir: PathBuf,
+
+    /// The directory where the repository will be written
+    #[structopt(short = "o", long = "outdir")]
+    outdir: PathBuf,
+
+    /// Follow symbolic links in the given directory when adding targets
+    #[structopt(short = "f", long = "follow")]
+    follow: bool,
+
+    /// Determins if entire repo should be signed
+    #[structopt(long = "sign-all")]
+    sign_all: bool,
+
+    // Use symlink instead of copying targets
+    #[structopt(short = "l", long = "link")]
+    link: bool,
+}
+
+impl UpdateTargetsArgs {
+    pub(crate) fn run(&self) -> Result<()> {
+        // load the repo
+        let datastore = tempdir().context(error::TempDir)?;
+        // We don't do anything with targets so we will use metadata url
+        let settings = tough::Settings {
+            root: File::open(&self.root).unwrap(),
+            datastore: &datastore.path(),
+            metadata_base_url: self.metadata_base_url.as_str(),
+            targets_base_url: self.metadata_base_url.as_str(),
+            limits: Limits::default(),
+            expiration_enforcement: ExpirationEnforcement::Safe,
+        };
```

Comment:
Messy diff, just suggesting the comment be moved to the site of the thing it's explaining.

```suggestion
        let settings = tough::Settings {
            root: File::open(&self.root).unwrap(),
            datastore: &datastore.path(),
            metadata_base_url: self.metadata_base_url.as_str(),
            // We don't do anything with targets so we will use metadata url
            targets_base_url: self.metadata_base_url.as_str(),
            limits: Limits::default(),
            expiration_enforcement: ExpirationEnforcement::Safe,
        };
```